packages/cpt-ui/src/components/prescriptionSearch/PrescriptionIdSearch.tsx

```diff
@@ -1,19 +1,199 @@
+            setLoading(false)
+            return
+        }
+
+        // Normalize and clean input
+        const cleaned = raw.replace(/[^a-zA-Z0-9+]/g, "").toUpperCase()
+
+        // Must be exactly 18 chars (excluding dashes)
+        if (cleaned.length !== 18) {
+            setErrorType("length")
+            setLoading(false)
+            return
+        }
+
+        const formatted = normalizePrescriptionId(cleaned)
+
+        // Validate full formatted pattern
+        const shortFormPattern = /^[0-9A-F]{6}-[0-9A-Z]{6}-[0-9A-F]{5}[0-9A-Z+]$/
+        if (!shortFormPattern.test(formatted)) {
+            setErrorType("noMatch")
+            setLoading(false)
+            return
+        }
+
+        const url = `${prescriptionDetailsEndpoint}/${formatted}`
+
+        // Fetch prescription details with auth headers. If successful, redirect to results.
+        // On failure, allow a known test ID through; otherwise, redirect to "not found".
+        try {
+            const response = await fetch(url, {
+                method: "GET",
+                headers: {
+                    Authorization: `Bearer ${auth?.idToken}`,
+                    "NHSD-Session-URID": "555254242106"
+                }
+            })
+
+            if (!response.ok) throw new Error(`Status Code: ${response.status}`)
+
+            navigate(`/prescription-results?prescriptionId=${formatted}`)
+        } catch (error) {
+            console.error("Error retrieving prescription details:", error)
+
+            // MOCK: Hardcoded response for known test ID
+            if (formatted === "C0C757-A83008-C2D93O") {
+                await new Promise((res) => setTimeout(res, 500)) // simulate loading
+                navigate(`/prescription-results?prescriptionId=${formatted}`)
+            } else {
+                navigate("/prescription-not-found?searchType=PrescriptionIdSearch")
+            }
+        } finally {
+            setLoading(false)
+        }
+    }
+
     return (
-        <>
-            <h1 data-testid="prescription-id-search-heading">Prescription ID Search</h1>
-            <div className="eps-modal-button-group" data-testid="prescription-id-search-button-group">
-                <Button
-                    className="nhsuk-button eps-modal-button"
-                    to="/prescription-results?prescriptionId=123456"
-                    data-testid="find-prescription-button"
-                >
-                    Find a prescription
-                </Button>
-            </div>
-        </>
+        <Container
+            className="nhsuk-width-container-fluid patient-search-form-container"
+            data-testid="prescription-id-search-container"
+        >
+            <Row>
+                <Col width="one-half">
+                    <Form onSubmit={handlePrescriptionDetails} noValidate>
+                        {errorType && (
+                            <div
+                                ref={errorRef}
```

Comment:
These should all be using the actual react components! https://main--65aa76b29d00a047fe683b95.chromatic.com/?path=/docs/form-elements-errorsummary--docs

This is why the behaviour is not as expected.

Reply:
I'll make the changes this afternoon